source/parser.hera

```diff
@@ -4424,18 +4428,21 @@ NestedJSXChildExpression
 ## Type Stuff
 
 TypeDeclaration
-  (TypeDeclarationModifier TrailingComment*)* TypeDeclarationRest -> { ts: true, children: $0 }
-
-TypeDeclarationModifier
-  "declare" NonIdContinue
-  Export
+  ( Export _? )? ( Declare _? )  TypeLexicalDeclaration
+  ( Export _? )? ( Declare _? )? TypeDeclarationRest -> { ts: true, children: $0 }
 
 TypeDeclarationRest
   TypeKeyword TrailingComment* IdentifierName TypeParameters? __ Equals __ Type
   Interface   TrailingComment* IdentifierName TypeParameters? InterfaceExtendsClause? InterfaceBlock
   Namespace   TrailingComment* IdentifierName NamespaceBlock
   FunctionSignature
```

Comment:
Originally I thought `class`es were missing here. But now I see that `TypeDeclarationRest` is for stuff that *must* be a type declaration, without any prefixes. Maybe add a comment?